### ISSUE TITLE
Restricting s3:PutObject in IAM policy doesn't work in UI

### ISSUE THREAD

**Issue by farhadh91** [minio/console#2838]
## NOTE
Hello, I am looking to restrict the putting of objects using the IAM policy. Therefore, I have added an IAM policy as follows:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:ListAllMyBuckets",
                "s3:GetObjectVersionTagging",
                "s3:ListBucket",
                "s3:DeleteObject",
                "s3:GetObject",
                "s3:GetObjectVersion"
            ],
            "Resource": [
                "arn:aws:s3:::mybucket/*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "s3:PutObject"
            ],
            "Resource": [
                "arn:aws:s3:::mybucket/*.jpg"
            ]
        }
    ]
}
```
## Expected Behavior
As per the policy, only users who are attached to the IAM can upload files with the .jpg extension, and this functionality is seamlessly working in the CLI.

## Current Behavior
Upon entering the user interface, the option to upload files is disabled for the user. Therefore, they will not be able to upload any files through the UI.

## Your Environment

* Version used (`minio --version`): RELEASE.2023-05-27T05-56-19Z
* Server setup and configuration: Kubernetes Cluster
* Operating System and version (`uname -a`):

**Comment by prakashsvmx** [comment]
`arn:aws:s3:::mybucket/*.jpg` i do not think such resource matching is supported in IAM policy. 


Please check https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-HTTPPOSTConstructPolicy.html#sigv4-PolicyConditions

**Comment by farhadh91** [comment]
Thanks for your response. It actually works in CLI and the users who are attached to this IAM policy, can't upload any file other than the jpg extension and they have got an error if the file has another extension!

**Comment by harshavardhana** [comment]
> Thanks for your response. It actually works in CLI and the users who are attached to this IAM policy, can't upload any file other than the jpg extension and they have got an error if the file has another extension!

It may work, but that is not something you should rely on. Since it is a hidden extension - it is not officially supported yet.

**Comment by farhadh91** [comment]
This is what AWS said and it works for me now. However, I have a question regarding the user interface. I noticed that when I apply this specific IAM policy, the upload files button becomes disabled. Is this a bug or is there a reason for this behavior?
https://repost.aws/knowledge-center/s3-allow-certain-file-types

<img width="1425" alt="Screenshot 2023-05-29 at 11 06 01 PM" src="https://github.com/minio/console/assets/37770395/f8b72b98-02e8-4bdc-96f9-b5967bd62375">

**Comment by harshavardhana** [comment]
> This is what AWS said and it works for me now. However, I have a question regarding the user interface. I noticed that when I apply this specific IAM policy, the upload files button becomes disabled. Is this a bug or is there a reason for this behavior? https://repost.aws/knowledge-center/s3-allow-certain-file-types
> 
> <img alt="Screenshot 2023-05-29 at 11 06 01 PM" width="1425" src="https://user-images.githubusercontent.com/37770395/241797242-f8b72b98-02e8-4bdc-96f9-b5967bd62375.png">

If AWS supports it and MinIO server also does, then we should allow this on Console UI.

It needs to be fixed since UI elements automatically evaluate the policy and disable buttons accordingly.

In your case the file types would need extra help. Such as the upload button must be visible and it must also use the resource constraint to show what all filetypes are indeed supported.

**Comment by farhadh91** [comment]
Thank you for looking into my previous issue. I have encountered another problem that I would like to address. 

I have successfully set the IAM policy for a user, and it is working as expected in both the UI and CLI. However, I am facing an issue while creating a new path or uploading a file in the UI. I am only able to do so when I am in the root path of the bucket. 

My requirement is to create a directory in another directory or upload a file to a specific directory. Unfortunately, the "Create new path" and "Upload File" buttons are disabled in the UI for non-root paths of the bucket. 

However, I am able to copy a directory or file to a non-root path of the bucket using the CLI.

![Screenshot 2023-07-30 at 1 55 04 PM](https://github.com/minio/console/assets/37770395/692e061d-a933-4ac9-9078-e06491f6394e)

![Screenshot 2023-07-30 at 1 55 12 PM](https://github.com/minio/console/assets/37770395/3d9e7ead-14c1-43ae-ba67-9f98bbd25885)
